frontends/bmd/src/pages/print_page.tsx

```diff
@@ -46,8 +47,11 @@ export function PrintPage(): JSX.Element {
     'precinctId is required to render PrintPage'
   );
   const printerTimer = useRef(0);
+  const printLock = useLock();
 
   const printBallot = useCallback(async () => {
+    /* istanbul ignore if */
+    if (!printLock.lock()) return;
```

Comment:
Curious why this lock wasn't needed when using the hold smartcard/auth hooks - were we seeing fewer re-renders with the prior hooks?

Reply:
Great question. The entire app re-renders every 3 seconds when the battery status is refreshed. That's been the case for a while. However, these re-renders weren't getting propagated down to this particular component (PrintPage) because they previously didn't update the BallotContext that this component accesses. This is because in the old code, all of the state and data put in the context was aggressively memoized to prevent it from changing on re-renders. 

The new useSmartcardAuth hook intentionally _does not_ memoize its result, because memoization is supposed to only be used to improve performance, not to ensure correctness. A component should be able to be re-rendered at any time without causing errant side-effects.

I'm hoping that this can help us flush out all of the places we're relying on memoization for correctness. My hunch is that we don't actually need to memoize much - we don't do much expensive rendering (besides ballots).